provider/resources.go

```diff
@@ -74,7 +74,7 @@ func applyResourceIDs(info tfbridge.PropertyVisitInfo) (tfbridge.PropertyVisitRe
 	}
 
 	// If the resource already has an ID property, just use that.
-	if _, ok := res.Schema.Schema().GetOk("id"); ok {
+	if id, ok := res.Schema.Schema().GetOk("id"); ok && id.Computed() && !id.Optional() {
```

Comment:
Can we get a regression test?

Reply:
Yes. Added in [c743e54](https://github.com/pulumi/pulumi-meraki/pull/21/commits/c743e54a6d9cdea695435fe9913d179fd5368213).